internal/telemetry/cluster_test.go

```diff
@@ -80,6 +80,276 @@ func TestK8sVersionRetrievesClusterVersion(t *testing.T) {
 	}
 }
 
+func TestAWSPlatformDeterminesOwnName(t *testing.T) {
+	t.Parallel()
+
+	c := newTestCollectorForClusterWithNodes(t, nodeAWS)
+	got, err := c.Platform(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := "aws"
+	if want != got {
+		t.Errorf("want %s, got %s", want, got)
+	}
+}
+
+func TestAzurePlatformDeterminesOwnName(t *testing.T) {
+	t.Parallel()
+
+	c := newTestCollectorForClusterWithNodes(t, nodeAzure)
+	got, err := c.Platform(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := "azure"
+	if want != got {
+		t.Errorf("want %s, got %s", want, got)
+	}
+}
+
+func TestGCPPlatformDeterminesOwnName(t *testing.T) {
+	t.Parallel()
+
+	c := newTestCollectorForClusterWithNodes(t, nodeGCP)
+	got, err := c.Platform(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := "gce"
```

Comment:
```suggestion
	want := "gcp"
```

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/660b340964b01bbf4367d0e3c00e2874fd54d8a1/cloud/providerid/providerid.go#L29

Reply:
value of the provider is taken from the ProviderID field set by the Cloud/Platform Provider. If it's `gcp` it will be reported in the Telemetry payload as `gcp`. We do not hard-code any values apart from `other` (used when ProviderID field is not present - populated by third party cloud/platform provider.